SPECS/keras/0001-Add-Keras-3.3.3.patch

```diff
@@ -0,0 +1,40 @@
+diff -urN keras-3.3.3/pip_build.py keras-3.3.3/pip_build.py
+--- keras-3.3.3/pip_build.py	2024-04-26 16:19:54.000000000 -0700
++++ keras-3.3.3/pip_build.py	2024-06-27 13:48:55.897838700 -0700
+@@ -29,6 +29,7 @@
+ package = "keras"
+ build_directory = "tmp_build_dir"
+ dist_directory = "dist"
++root_path = ""
+ to_copy = ["setup.py", "README.md"]
+ 
+ 
+@@ -86,7 +87,8 @@
+         move_tf_keras_directory()
+         print(os.getcwd())
+ 
+-        from keras.src.version import __version__  # noqa: E402
++        #from keras.src.version import __version__  # noqa: E402

```

Comment:
was getting an error so just hardcoded this instead

Reply:
What was the error?  Can you comment in the SPEC file.  And if we absolutely need this, can you add something to the %build section that fails the build when somebody upgrades keras to anything other than 3.3.3?

Reply:
The error was it would complain about missing tensorflow module.  Tensorflow used to build time dep but with Keras 3.0 not anymore. Keras also doesn't use bazel anymore either.  
Currently, TensorFlow is not even runtime nor a build time dependency of keras. Keras is runtime req of tensorflow.  "pip3 install keras" doesn't install tensorflow.  Hence hardcoded this. If not it would have created TensorFlow as a build time req just to get the version number. 
I will add a variable.

Reply:
here's the stack trace. 
<img width="550" alt="image" src="https://github.com/user-attachments/assets/6f672ce3-22c1-45c3-a3b7-c1a4d677d8a2">